modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -432,9 +451,12 @@ interface GetCurrentComponentHandler {
  *
  * @returns {object} The current component as JSON.
  */
-export function getComponent(): object {
+export function getComponent<
+  Config extends object = object,
+  Regions extends Record<string,Region> = Record<string,Region>
+>(): Component<Config,Regions> {
     const bean = __.newBean<GetCurrentComponentHandler>('com.enonic.xp.lib.portal.current.GetCurrentComponentHandler');
-    return __.toNativeObject(bean.execute());
+    return __.toNativeObject(bean.execute()) as Component<Config,Regions>;
```

Comment:
I suggest  to change output type in `GetCurrentComponentHandler` for the `execute` function to `Component<Config,Regions>`. To avoid type casting here.